examples/fluids/src/setupdm.c

```diff
@@ -97,11 +97,12 @@ PetscErrorCode SetUpDM(DM dm, ProblemData *problem, PetscInt degree,
     CHKERRQ(ierr);
     ierr = PetscFEDestroy(&fe); CHKERRQ(ierr);
   }
-  {
-    // Empty name for conserved field (because there is only one field)
-    PetscSection section;
-    ierr = DMGetLocalSection(dm, &section); CHKERRQ(ierr);
-    ierr = PetscSectionSetFieldName(section, 0, ""); CHKERRQ(ierr);
+
+  // Empty name for conserved field (because there is only one field)
+  PetscSection section;
+  ierr = DMGetLocalSection(dm, &section); CHKERRQ(ierr);
+  ierr = PetscSectionSetFieldName(section, 0, ""); CHKERRQ(ierr);
+  if (!phys->primitive) {
```

Comment:
Stylistically, I think it's more clear to say `if (cond) {} else {}` rather than `if (!cond) {} else {}` when there isn't a strong reason otherwise.